include/tdycore.h

```diff
@@ -77,19 +77,18 @@ PETSC_EXTERN PetscErrorCode TDyInit(int argc, char* argv[]);
 PETSC_EXTERN PetscErrorCode TDyInitNoArguments(void);
 PETSC_EXTERN PetscErrorCode TDyFinalize(void);
 
-PETSC_EXTERN PetscErrorCode TDyCreate(TDy *tdy);
-PETSC_EXTERN PetscErrorCode TDySetMode(TDy tdy, TDyMode mode);
-PETSC_EXTERN PetscErrorCode TDySetDiscretizationMethod(TDy tdy,
-                                                       TDyMethod method);
-PETSC_EXTERN PetscErrorCode TDySetDM(TDy tdy,DM dm);
-PETSC_EXTERN PetscErrorCode TDySetFromOptions(TDy tdy);
-PETSC_EXTERN PetscErrorCode TDySetupNumericalMethods(TDy tdy);
+PETSC_EXTERN PetscErrorCode TDyCreate(TDy*);
```

Comment:
I'm not sure we want to get rid of these parameter names for functions that belong to TDycore's public API. If we adopt a Doxygen-style approach to annotating source code, we would describe these parameter names in the annotations. Of course, we can annotate the implementations instead of the declarations, but many folks are used to browsing header files to see what a library can do.

Reply:
I'm here following PETSc's developer manual. Another reason for us to have our own style guide.

Reply:
Sounds like we need a style guide!

Reply:
The rationale for the PETSc style guide is that the compiler doesn't check the names appearing here so they can become outdated and thus misleading.

Reply:
That makes sense for an approach that doesn't annotate header files. If we decide to annotate our functions in their implementations instead of headers, then this approach is fine.